src/main/java/uk/ac/ebi/eva/pipeline/io/mappers/VariantVcfFactory.java

```diff
@@ -329,40 +268,48 @@ protected void parseSplitSampleData(Variant variant, VariantSource source, Strin
     }
 
     /**
-     * Checks whether a sample should be included in a variant's list of
-     * samples. If current allele index is not found in the genotype and not all
-     * alleles are references/missing, then the sample must not be included.
+     * If this is a field other than the genotype (GT), return unmodified.
+     *
+     * If this field is the genotype (GT) then, intern it into the String pool to avoid storing lots of "0/0". In case
+     * that the variant is multiallelic and we are currently processing one of the secondary alternates (alleleIdx >=1),
+     * change the allele codes to represent the current alternate as allele 1. {@see mapToMultiallelicIndex}
      *
-     * @param genotype The genotype
-     * @param alleleIdx The index of the allele
-     * @return If the sample should be associated to the variant
+     * @param alleleIdx current alternate being processed. 1 for first alternate, 2 or more for a secondary alternate.
+     * @param formatField as shown in the FORMAT column. most probably the GT field.
+     * @param sampleField parsed value in a column of a sample, such as a genotype, e.g. "0/0".
+     * @return
      */
-    private boolean shouldAddSampleToVariant(String genotype, int alleleIdx) {
-        if (genotype.contains(String.valueOf(alleleIdx))) {
-            return true;
-        }
-        
-        if (!genotype.contains("0") && !genotype.contains(".")) {
-            return false;
-        }
-        
-        String[] alleles = genotype.split("[/|]");
-        for (String allele : alleles) {
-            if (!allele.equals("0") && !allele.equals(".")) {
-                return false;
+    private String processSampleField(int alleleIdx, String formatField, String sampleField) {
+        if (formatField.equalsIgnoreCase("GT")) {
+            if (alleleIdx >= 1) {
+                Genotype genotype = new Genotype(sampleField);
```

Comment:
if the alleles are numbers or ".", the class does little more than checking if it is "." or a number. I don't think there's much to optimize there.